lib/kafka/connection.rb

```diff
@@ -196,6 +181,21 @@ def read_response(response_class, notification)
       raise
     end
 
+    # Kafka brokers automatically close client connections after a period of
+    # inactivity. If this has happened, it's safe to re-open the connection
+    # and retry the request.
+    def with_retry
+      yield
+    rescue Errno::EPIPE, Errno::ECONNRESET, EOFError
```

Comment:
I'd assume that we'd get the same errors – my understanding is that the SSL "socket" is really just two normal sockets under the hood.